crates/engine/tree/src/tree/mod.rs

```diff
@@ -518,7 +518,10 @@ where
         loop {
             match self.try_recv_engine_message() {
                 Ok(Some(msg)) => {
-                    self.on_engine_message(msg);
+                    if let Err(fatal) = self.on_engine_message(msg) {
+                        error!(target: "engine", %fatal, "insert block fatal error");
+                        // return
```

Comment:
If it breaks it'll just loop again

and

If it returns we bail out of run()?

I'm leaning towards break but I'm not sure which behavior we want on `InsertBlockFatalError`

Reply:
> If it breaks it'll just loop again

only `continue` will continue to loop again, `break` here should be the same as `return`, since there is only one level of loop

Reply:
ah okay thanks for clarifying